### ISSUE TITLE
Needs to be migrated for web support

### ISSUE THREAD

**Issue by awvreenen95** [dint-dev/universal_html#95]
I am attempting to compile a webassembly build for a project I'm working on that makes use of mqtt. The mqtt_client library is dependent on universal_html which does not support webassembly since the dart:html library is not supported.

Please see some more details in an issue I raised [on flutter repo](https://github.com/flutter/flutter/issues/143157) and also an issue raised on the [mqtt_client repo](https://github.com/shamblett/mqtt_client/issues/503).

**Comment by Macacoazul01** [comment]
Here we're having this error when building for flutter 3.19: 
`Error: The type 'JSObject' can't be used as supertype.`

**Comment by navaronbracke** [comment]
We should really deprecate this library in favor of `package:web` that is vendored by the Dart team itself.

https://pub.dev/packages/web

**Comment by NotTsunami** [comment]
> We should really deprecate this library in favor of `package:web` that is vendored by the Dart team itself.
> 
> https://pub.dev/packages/web

Yes, specifically in the context of MQTT, they should revert d7eec87d5f151e09d006fab73a68346202d31719, and use [conditional imports](https://dart.dev/interop/js-interop/package-web#conditional-imports) to control rather or not `package:web` is used.

**Comment by rayliverified** [comment]
@schultek released a `universal_web` package.
https://pub.dev/packages/universal_web

Maybe @dint-dev could be invited to help maintain it if they're willing.
